automation/jleechanorg_pr_automation/jleechanorg_pr_monitor.py

```
@@ -775,6 +809,7 @@ def _get_head_commit_details(
         owner, name = repo_full_name.split("/", 1)
 
         # Validate GitHub naming constraints (alphanumeric, hyphens, periods, underscores, max 100 chars)
+        import re
```

Ruff (0.14.7)

`import` should be at the top-level of a file

(PLC0415)